docs/quant/pyquant/ssvi.py

```diff
@@ -0,0 +1,238 @@
+from typing import Union
+
+import numba as nb
+import numpy as np
+from scipy.optimize import minimize
+
+from .black_scholes import *
+from .common import *
+from .svi import *
+from .svi import Rho
+from .vol_surface import *
+
+
+@nb.experimental.jitclass([("delta_param", nb.float64)])
+class DeltaParam:
+    def __init__(self, delta_param: nb.float64):
+        self.delta_param = delta_param
+
+
+@nb.experimental.jitclass([("mu", nb.float64)])
+class Mu:
+    def __init__(self, mu: nb.float64):
+        self.mu = mu
+
+
+@nb.experimental.jitclass([("theta", nb.float64)])
+class Theta:
+    def __init__(self, theta: nb.float64):
+        if not (theta >= 0):
+            raise ValueError("Theta not >= 0")
+        self.theta = theta
+
+
+@nb.experimental.jitclass([("zeta", nb.float64)])
+class Zeta:
+    def __init__(self, zeta: nb.float64):
+        if not (zeta > 0):
+            raise ValueError("Zeta not > 0")
+        self.zeta = zeta
+
+
+@nb.experimental.jitclass([("lambda_", nb.float64)])
+class Lambda:
+    def __init__(self, lambda_: nb.float64):
+        if not (lambda_ >= 0):
+            raise ValueError("Lambda not >= 0")
+        self.lambda_ = lambda_
+
+
+@nb.experimental.jitclass([("eta", nb.float64)])
+class Eta:
+    def __init__(self, eta: nb.float64):
+        if not (eta >= 0):
+            raise ValueError("Eta not >= 0")
+        self.eta = eta
+
+
+@nb.experimental.jitclass([("beta", nb.float64)])
+class Beta:
+    def __init__(self, beta: nb.float64):
+        if not (beta >= 0):
+            raise ValueError("Beta not >= 0")
+        self.beta = beta
+
+
+@nb.experimental.jitclass([("alpha", nb.float64)])
+class Alpha:
+    def __init__(self, alpha: nb.float64):
+        self.alpha = alpha
+
+
+@nb.experimental.jitclass([("gamma_", nb.float64)])
+class Gamma_:
+    def __init__(self, gamma_: nb.float64):
+        self.gamma_ = gamma_
+
+
+@nb.experimental.jitclass(
+    [
+        ("delta_param", nb.float64),
+        ("mu", nb.float64),
+        ("rho", nb.float64),
+        ("theta", nb.float64),
+        ("zeta", nb.float64),
+        ("lambda_", nb.float64),
+        ("eta", nb.float64),
+        ("beta", nb.float64),
+        ("alpha", nb.float64),
+        ("gamma_", nb.float64),
+    ]
+)
+class SVINaturalParams:
+    def __init__(
+        self, delta_param: DeltaParam, mu: Mu, rho: Rho, theta: Theta, zeta: Zeta
+    ):
+        self.delta_param = delta_param.delta_param
+        self.mu = mu.mu
+        self.rho = rho.rho
+        self.theta = theta.theta
+        self.zeta = zeta.zeta
+
+    def array(self) -> nb.float64[:]:
+        return np.array([self.delta_param, self.mu, self.rho, self.theta, self.zeta])
+
+
+class SSVI:
+    def __init__(
+        self,
+        vol_smile_chain_spaces: list[VolSmileChainSpace],
```

Comment:
the vol surface argument should be an argument in `calibrate` method. You should use the `VolSurfaceDeltaSpace` [class](https://github.com/grinisrit/noa/blob/118dfdf7f339492fa8ee2eca1b635b6f0982b1b0/docs/quant/pyquant/vol_surface.py#L285)